server/channels/store/sqlstore/shared_channel_store.go

```diff
@@ -398,8 +403,10 @@ func sharedChannelRemoteFields(prefix string) []string {
 		prefix + "IsInviteAccepted",
 		prefix + "IsInviteConfirmed",
 		prefix + "RemoteId",
+		prefix + "LastPostCreateAt",
+		"COALESCE(" + prefix + "LastPostCreateID,'') AS LastPostCreateID",
 		prefix + "LastPostUpdateAt",
-		prefix + "LastPostId",
+		"COALESCE(" + prefix + "LastPostId,'') AS LastPostUpdateID",
```

Comment:
`LastPostCreateID` is a new field that is not being back-filled.  At some point in my dev testing I ended up with a null LastPostID.  It is a null-able field, because it was added in a migration after table creation and not back-filled.

Reply:
Ah, got it, so maybe the question is: why is this not being backfilled, when we're doing it for the other new columns?
```
ALTER TABLE RemoteClusters ADD COLUMN IF NOT EXISTS PluginID VARCHAR(190) NOT NULL DEFAULT '';

ALTER TABLE SharedChannelRemotes ADD COLUMN IF NOT EXISTS LastPostCreateAt bigint NOT NULL DEFAULT 0;

ALTER TABLE SharedChannelRemotes ADD COLUMN IF NOT EXISTS LastPostCreateID VARCHAR(26);
```

Reply:
@lieut-data the `RemoteClusters` table is always going to be small, however the `SharedChannelRemotes` table can get large, especially when we start sharing all channels.  That table gets a record per per per remote.  My understanding is that adding a column with no-null and default in older MySQL requires a re-write of the table.

Reply:
That's true, and a mistake on my part.  Now I'm second guessing if I should be adding the default to the other columns. The migration guide from server team seems to suggest adding nullable columns is fine, but doesn't explicitly state non-null with defaults is not.  Looking at MySQL documentation, there is a table re-write for older versions.  Despite EOL on MySQL versions, I don't think we can just will customers to upgrade on our timeline.  I'll use coalesce for now and not create migration headaches for folks straggling on DB upgrades.

Reply:
@lieut-data looking at other migrations, I do see columns added with defaults, including the reactions table.  I'm now second guessing whether I can use defaults or not.

Reply:
You can see the table here: https://dev.mysql.com/doc/refman/5.7/en/innodb-online-ddl-operations.html#online-ddl-column-operations. Adding a column rebuilds the table, but it permits concurrent DML. That's the most important part. We don't care how long it takes, as long as it does not block the table. 

We can also easily verify by adding a new column with default on the posts table and trying to make some queries and see if they return.

Reply:
Thanks for the detail agnivade.  I'll keep that in mind for future schema changes.